### DISCUSSION TITLE
Anyway to skip relocation or not fix-up some files?

### DISCUSSION THREAD

**Discussion by Gcenx** [Homebrew/discussions#3597] [Tap maintenance and Homebrew development]
If relocation can only be disabled completely that’s fine, just then would there also be a correct way to skip bottling?

**Comment by SMillerDev** [comment]
It can be avoided by building everything from source. There's no way to partially use bottles.

**Reply by Gcenx** [reply]
Maybe I didn't make that part clear I was referring to the relocation not bottling as (I plan to skip bottling)

**Comment by carlocab** [comment]
When do you want to skip relocation? Is it when pouring a bottle, or when creating one?

And when do you want to skip bottling? Is this from using `brew test-bot`?

**Reply by Gcenx** [reply]
> When do you want to skip relocation? Is it when pouring a bottle, or when creating one?

Might be safer to completely skip relocation.

> And when do you want to skip bottling? Is this from using brew test-bot?

Yes I'd like to have the formula still pass test-bot and not be a problem later, guessing just letting it pass and not using `pr-pull` would be right?

**Reply by carlocab** [reply]
Actually, it suffices to skip relocation when creating a bottle. If relocation is not done when creating it, then there is nothing to relocate upon pouring it.

You might then be interested in the `--skip-relocation` flag of `brew bottle` and `brew test-bot`. Not 100% certain this does what you're looking for, but it sounds promising.

Doing `brew test-bot --skip-relocation` merely passes `--skip-relocation` to `brew bottle`. At least, that's what it does in theory. That flag is not tested in our CI, so it may very well not work at all. PRs to fix it welcome if it doesn't work.

**Reply by carlocab** [reply]
To skip building bottles, pass `--build-from-source` to `brew test-bot`.
